src/main/java/com/sprint/mission/discodeit/repository/file/FileUserRepository.java

```diff
@@ -0,0 +1,170 @@
+package com.sprint.mission.discodeit.repository.file;
+
+import com.sprint.mission.discodeit.entity.Channel;
+import com.sprint.mission.discodeit.entity.Message;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.entity.UserStatus;
+import com.sprint.mission.discodeit.repository.UserRepository;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class FileUserRepository implements UserRepository, Serializable {
+
+    private static final String USER_FILE_PATH = "./data/user.ser";
+    private static final FileUserRepository instance = new FileUserRepository();
+
+    public FileUserRepository() {}
+    public static FileUserRepository getInstance() {
+        return instance;
+    }
+
+    public List<User> loadUsers() {
+        File file = new File(USER_FILE_PATH);
+
+        if (!file.exists() || file.length() == 0) {
+            return new ArrayList<User>(); // 해당 파일이 없거나 비어 있다면 빈 ArrayList를 반환
```

Comment:
```suggestion
           return Collections.emptyList();// 해당 파일이 없거나 비어 있다면 빈 ArrayList를 반환
```
빈리스트를 반환하는게 목적이라면 의도가 명확하게 위처럼 표현하시는게 훨씬 직관적입니다.
불변 리스트를 리턴하는 것이기도 하구요.